From 8f2e1f9f3bcbc92c22c30d32dc1cfd228dc57de6 Mon Sep 17 00:00:00 2001
From: Hariharan Basuthkar <quic_hbasuthk@quicinc.com>
Date: Thu, 22 Feb 2024 16:06:34 +0530
Subject: [PATCH] wifi: cfg80211: Send power-mode in get_interface command

In order to display the power mode in the "iw dev" command,
send the power in the NL80211_CMD_GET_INTERFACE.

Signed-off-by: Hariharan Basuthkar <quic_hbasuthk@quicinc.com>
---
 net/wireless/nl80211.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/net/wireless/nl80211.c b/net/wireless/nl80211.c
index 48fe770..b299ae8 100644
--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@ -4278,6 +4278,11 @@ static int nl80211_send_iface(struct sk_buff *msg, u32 portid, u32 seq, int flag
 		goto nla_put_failure;
 
 	wdev_lock(wdev);
+
+	if (nla_put_u8(msg, NL80211_ATTR_6G_REG_POWER_MODE,
+		       wdev->reg_6g_power_mode))
+		goto nla_put_failure_locked;
+
 	switch (wdev->iftype) {
 	case NL80211_IFTYPE_AP:
 	case NL80211_IFTYPE_P2P_GO:
-- 
2.34.1

